ui-lib/src/main/java/co/electriccoin/zcash/ui/common/FlowKtx.kt

```diff
@@ -0,0 +1,22 @@
+@file:Suppress("ktlint:filename")
+
+package co.electriccoin.zcash.ui.common
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.WhileSubscribed
+import kotlinx.coroutines.flow.stateIn
+
+/**
+ * Converts Flow to StateFlow with default sharing strategy and initial value set to null
+ */
+fun <T> Flow<T>.toStateFlow(
```

Comment:
I'd prefer we not introduce this as a one-off.  We either need to use this everywhere (thus modifying all our view models) or nowhere.

I think there's value in keeping our existing pattern of being explicit with:

```
.stateIn(
        viewModelScope,
        SharingStarted.WhileSubscribed(ANDROID_STATE_FLOW_TIMEOUT),
        null
    )
```

The reason I'm not 100% convinced this syntactic sugar is the right move is that some places we specifically use different behaviors like `WalletCoordinator` and `SupportViewModel`.  I think being explicit makes it obvious that someone really thought through the usage of the StateFlow.

Thoughts?

Reply:
Yes i agree with what you saying, but i see this extension just as a nice way to remove unwanted boilerplate.  On the other hand i think you're right that being explicit makes it more obvious.